src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -697,13 +700,26 @@ private IOperation CreateBoundConversionOperation(BoundConversion boundConversio
                 }
 
                 Lazy<IOperation> operand = new Lazy<IOperation>(() => Create(boundConversion.Operand));
+                ITypeSymbol type = boundConversion.Type;
+                Optional<object> constantValue = ConvertToOptional(boundConversion.ConstantValue);
+
+                // If this is a lambda conversion, we need to check to see if it's a delegate conversion.
+                if (boundConversion.Operand.Kind == BoundKind.Lambda ||
+                    boundConversion.Operand.Kind == BoundKind.UnboundLambda)
+                {
+                    // We need to determine if this is a conversion to a delegate type. If it is, it's a delegate creation, so we
+                    // return an IDelegateCreationExpression instead of a conversion.
+                    if (boundConversion.Type.InheritsSpecialType(SpecialType.System_Delegate))
+                    {
+                        return new LazyDelegateCreationExpression(operand, _semanticModel, syntax, type, constantValue, isImplicit);
```

Comment:
>return new LazyDelegateCreationExpression(operand, _semanticModel, syntax, type, constantValue, isImplicit); [](start = 24, length = 108)

Do we want to get here for error cases? Do we have tests hitting this code path?

Reply:
Do we have tests hitting this code path for error cases?

---
In reply to: [138489201](https://github.com/dotnet/roslyn/pull/22056#discussion_r138489201) [](ancestors = 138489201)

Reply:
We do. All of the invalid Lambda tests hit this path.